eclipse: add Eclipse IDE for Java Developers 4.7 and 4.8

joakim1999 opened the pull request:
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

joakim1999 committed:
eclipse: add Eclipse IDE for Java Developers 4.7 and 4.8

rycee commented on pkgs/applications/editors/eclipse/default.nix:
```diff
@@ -227,6 +227,42 @@ rec {
+  
+  ### Eclipse Java
+  
+  eclipse-java-47 = buildEclipse {
```
Since this is a new package and there is no concern about accommodating people preferring to stay on an older version I think we can remove the 4.7 version here.

joakim1999 replied:
Ok

rycee commented on pkgs/applications/editors/eclipse/default.nix:
```diff
@@ -227,6 +227,42 @@ rec {
+      else throw "Unsupported system: ${stdenv.system}";
+  };
+  
+  eclipse-java-48 = buildEclipse {
```
Should add an `eclipse-java = eclipse-java-48;` entry for people that always want the latest version.

rycee replied:
Actually, isn't the 2018-09 release called 4.9.0? If so then the package and attribute names should be updated to match.

joakim1999 replied:
Don't know how i missed that. I am gonna try out that version on NixOS and add the newest version.

rycee commented on pkgs/applications/editors/eclipse/default.nix:
```diff
@@ -227,6 +227,42 @@ rec {
+  
+  eclipse-java-48 = buildEclipse {
+    name = "eclipse-java-4.8.0";
+    description = "Eclipse IDE for Java Developers, Oxygen release";
```
Can remove the ", Oxygen release" part.

joakim1999 replied:
I forgot to change it to Photon release. But it's ok if i just don't say anything about the release?

rycee replied:
Yeah, I think that should be fine. Alternatively, if you want, then put
```
description = "Eclipse IDE for Java Developers 2018-09"
```
similar to eclipse-platform-49 and eclipse-sdk-49.

rycee commented:
Thanks for the contribution! Looks good overall, I just added a few minor comments.

joakim1999 committed:
eclipse: fixed incorrect version
Changed entry eclipse-java-48 to eclipse-java-49 to match the files represented inside the entry.
Added eclipse-java entry pointing to latest version (eclipse-java-49)

rycee commented:
@grahamcofborg build eclipses.eclipse-java

GrahamcOfBorg commented:
<!--REQUEST_ID=25bddeac-c876-4144-aeed-18472fb78b4b-->
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48395&attempt_id=e9015a5f-20fd-449a-91c7-63ac3415bb15)

The following builds were skipped because they don't evaluate on aarch64-linux: eclipses.eclipse-java

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnfree = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnfree = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=25bddeac-c876-4144-aeed-18472fb78b4b-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48395&attempt_id=20656ae6-647b-4780-bb60-ebf4dd08abef)

Attempted: eclipses.eclipse-java

<details><summary>Partial log (click to expand)</summary><p>

```
eclipse/readme/
eclipse/readme/readme_eclipse.html
eclipse/dropins/
eclipse/eclipse.ini
eclipse/configuration/
eclipse/configuration/org.eclipse.equinox.simpleconfigurator/
eclipse/configuration/org.eclipse.equinox.simpleconfigurator/bundles.info
eclipse/configuration/config.ini
eclipse/configuration/org.eclipse.update/
eclipse/configuration/org.eclipse.update/platform.xml
```
</p></details>

rycee commented:
Squashed and rebased to master in 6fe0f07557e481a4861ee938897d25e219dced1c. Thanks!
